{"cells": [{"metadata": {}, "cell_type": "code", "source": "pwd\n", "execution_count": 1, "outputs": [{"output_type": "execute_result", "execution_count": 1, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "!pip install keras==2.2.4\n!pip install tensorflow==1.14.0", "execution_count": 2, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: keras==2.2.4 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (2.2.4)\nRequirement already satisfied: numpy>=1.9.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.20.3)\nRequirement already satisfied: pyyaml in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (5.4.1)\nRequirement already satisfied: scipy>=0.14 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.7.3)\nRequirement already satisfied: keras-preprocessing>=1.0.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.1.2)\nRequirement already satisfied: keras-applications>=1.0.6 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.0.8)\nRequirement already satisfied: h5py in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (3.2.1)\nRequirement already satisfied: six>=1.9.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.15.0)\n\u001b[31mERROR: Could not find a version that satisfies the requirement tensorflow==1.14.0 (from versions: 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.7.0rc0, 2.7.0rc1, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.8.0rc0, 2.8.0rc1, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.9.0rc0, 2.9.0rc1, 2.9.0rc2, 2.9.0, 2.9.1, 2.9.2, 2.10.0rc0, 2.10.0rc1, 2.10.0rc2, 2.10.0rc3, 2.10.0, 2.11.0rc0, 2.11.0rc1, 2.11.0rc2)\u001b[0m\n\u001b[31mERROR: No matching distribution found for tensorflow==1.14.0\u001b[0m\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "from keras.models import Sequential#api,se,\nfrom keras.layers import Dense #ADD LAYERS\nfrom keras.layers import Convolution2D #CON\nfrom keras.layers import MaxPooling2D #\nfrom keras.layers import Flatten\n", "execution_count": 3, "outputs": [{"output_type": "stream", "text": "Using TensorFlow backend.\n", "name": "stderr"}]}, {"metadata": {}, "cell_type": "code", "source": "from keras.preprocessing.image import ImageDataGenerator\ntrain_datagen = ImageDataGenerator(resacale=1./255,shear_range = 0.2,zoom_range = 0.2,horizontal_flip = True)\ntest_datagen = ImageDataGenerator(rescale = 1./255)\n                                ", "execution_count": 4, "outputs": [{"output_type": "error", "ename": "TypeError", "evalue": "__init__() got an unexpected keyword argument 'resacale'", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_1217/3576486749.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mkeras\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpreprocessing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mimage\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mImageDataGenerator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mtrain_datagen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mImageDataGenerator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresacale\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1.\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m255\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mshear_range\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0.2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mzoom_range\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0.2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mhorizontal_flip\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mtest_datagen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mImageDataGenerator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrescale\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1.\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m255\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;31mTypeError\u001b[0m: __init__() got an unexpected keyword argument 'resacale'"]}]}, {"metadata": {}, "cell_type": "code", "source": "\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='jsk7HE-RUJkiT9aTbPw8s0cxcdV-DG_Gu8p96m2UfAX4',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'emergingmethodsforearlydetectiono-donotdelete-pr-mprzel1nrl7ovu'\nobject_key = 'archive.zip.rar'\n\nstreaming_body_2 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='jsk7HE-RUJkiT9aTbPw8s0cxcdV-DG_Gu8p96m2UfAX4',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'emergingmethodsforearlydetectiono-donotdelete-pr-mprzel1nrl7ovu'\nobject_key = 'Image_preprocessing.ipynb'\n\nstreaming_body_1 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "from io import BytesIO \nimport zipfile\nunzip = zipfile.ZipFile(BytesIO(streaming_body_1.read()),'r')\nfile_paths = unzip.namelist()\nfor path in file_paths :\nunzip.extract(paths)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "pwd\n", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "import os\nfilename = os.listdir('/home/wsuser/work/archive/train_set')", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "x_train = train_datagen.flow_from_directory(\"/home/wsuser/work/archive/train_set\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='jsk7HE-RUJkiT9aTbPw8s0cxcdV-DG_Gu8p96m2UfAX4',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'emergingmethodsforearlydetectiono-donotdelete-pr-mprzel1nrl7ovu'\nobject_key = 'train_set.rar'\n\nstreaming_body_4 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n\",target_size=(64,64),batch_size= 5,color_mode='rgb',class_mode='sparse')\nx_test  =  test_datagen.flow_from_directory(\"/home/wsuser/work/archive/test_set\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='jsk7HE-RUJkiT9aTbPw8s0cxcdV-DG_Gu8p96m2UfAX4',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'emergingmethodsforearlydetectiono-donotdelete-pr-mprzel1nrl7ovu'\nobject_key = 'test_set.rar'\n\nstreaming_body_3 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n\",target_size=(64,64),batch_size= 5,color_mode='rgb',class_mode='sparse')", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "x_train.class_indices", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='jsk7HE-RUJkiT9aTbPw8s0cxcdV-DG_Gu8p96m2UfAX4',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'emergingmethodsforearlydetectiono-donotdelete-pr-mprzel1nrl7ovu'\nobject_key = 'model_building_.ipynb'\n\nstreaming_body_5 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "model.save(\"archive.h5\")", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "!tar -zcvf image-classification-model_new.tgz archive.h5", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "ls -1", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "!pip install watson-machine-learning-client --upgrade", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "from ibm_watson_machine_learning import APIclient\nwml_crenditals = {\n                  \"url\": \"https://us-south.ml.cloud.ibm.com\",\n                   \"apikey\":\"vvv-7YgeBvirfZ20lyrFVrHJT9J9MAVQRRenEd79erDtFy\"\n}\nClient = APIclient(wml_credentials)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "Client = APIclient(wml_credentials)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "def  guid_from_space_name(Client,space_name):\n    space = Client.spaces.get_details()\n    #print(space)\n    #return(next(item for item in space['resources'] if item['entity'][\"name\"] == space_name)['metadata']['id'])", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "space_uid = guid_from_space_name(Client,'imageclassification')\nprint(\"Space UID = \" + space_uid)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "Client.set.default_space(space_uid)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "Client.repositary.download(\"vvv-7YgeBvirfZ20lyrFVrHJT9J9MAVQRRenEd79erDtFy\",'archive.tar.gz')", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "", "execution_count": null, "outputs": []}], "metadata": {"kernelspec": {"name": "python3", "display_name": "Python 3.9", "language": "python"}, "language_info": {"name": "python", "version": "3.9.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat": 4, "nbformat_minor": 1}